### PR TITLE
Add validations for SubnetPort CRD

### PR DESCRIPTION
**Pull request by lxiaopei**
Add validations for SubnetPort CR to specify one of subnet or subnetSet under spec.

### CODE REVIEW

**Comment by lxiaopei** [comment]
e2e

**Comment by yanjunz97** [comment]
Shall we remove the check in subnetport controller for this validation?
https://github.com/vmware-tanzu/nsx-operator/blob/bf1880a2327f379c4a1f2331c3690aa4387653de/pkg/controllers/subnetport/subnetport_controller.go#L82-L87

**Comment by lxiaopei** [comment]
> Shall we remove the check in subnetport controller for this validation?
> 
> https://github.com/vmware-tanzu/nsx-operator/blob/bf1880a2327f379c4a1f2331c3690aa4387653de/pkg/controllers/subnetport/subnetport_controller.go#L82-L87

thanks，removed.

**Review by dantingl — APPROVED** [review]

**Comment by yanjunz97** [comment]
Fix the UT?

**Review by yanjunz97 — APPROVED** [review]

File: build/yaml/crd/vpc/crd.nsx.vmware.com_subnetports.yaml [hunk 1]
```diff
@@ -59,6 +59,10 @@ spec:
                 description: SubnetSet defines the parent SubnetSet name of the SubnetPort.
                 type: string
             type: object
+            x-kubernetes-validations:
+            - message: Only one of subnet or subnetSet can be specified
+              rule: has(self.subnet) && !has(self.subnetSet) || !has(self.subnet)
```
Comment (dantingl) [inline build/yaml/crd/vpc/crd.nsx.vmware.com_subnetports.yaml:64]: This rule does not include both subnet and subnetset empty case

Comment (lxiaopei) [reply build/yaml/crd/vpc/crd.nsx.vmware.com_subnetports.yaml:64]: if both subnet and subnetset empty,  has(self.subnet) && !has(self.subnetSet) || !has(self.subnet) && has(self.subnetSet) cannot be true.

**Review by yanjunz97 — DISMISSED** [review]
LGTM

**Review by dantingl — APPROVED** [review]

**Pull request merged**